File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDONoticeMessageParamModel.kt
//
//  IDONoticeMessageParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
open class IDONoticeMessageParamModel(
    evtType: Int,
    msgID: Int,
    supportAnswering: Boolean,
    supportMute: Boolean,
    supportHangUp: Boolean,
    contact: String,
    phoneNumber: String,
    dataText: String
) : IDOBaseModel {

    /// Protocol library version number
    /// Default version=1
    /// Version=2 is the sent format with msg_id
    @SerializedName("verison")
    private var verison: Int = 1

    /// Message application type
    @SerializedName("evt_type")
    var evtType: Int = evtType

    /// Message ID
    /// If evt_type is message reminder, mesg_ID is valid
    @SerializedName("msg_id")
    var msgID: Int = msgID

    /// Support answering: 1
    /// Do not support answering: 0
    @SerializedName("support_answering")
    var supportAnswering: Boolean = supportAnswering

    /// Support mute: 1
    /// Do not support mute: 0
    @SerializedName("support_mute")
    var supportMute: Boolean = supportMute

    /// Support hang up: 1
    /// Do not support hang up: 0
    @SerializedName("support_hang_up")
    var supportHangUp: Boolean = supportHangUp

    /// Contact name (maximum 63 bytes)
    @SerializedName("contact")
    var contact: String = contact

    /// Phone number (maximum 31 bytes)
    @SerializedName("phone_number")
    var phoneNumber: String = phoneNumber

    /// Message content (maximum 249 bytes)
    @SerializedName("data_text")
    var dataText: String = dataText


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDONoticeMessageParamModel.swift
//
//  IDONoticeMessageParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/17.
//

import Foundation

@objcMembers
public class IDONoticeMessageParamModel: NSObject, IDOBaseModel {
    /// Protocol library version number
    /// Default version=1
    /// Version=2 is the sent format with msg_id
    private var verison: Int
    /// Message application type
    public var evtType: Int
    /// Message ID
    /// If evt_type is message reminder, mesg_ID is valid
    public var msgID: Int
    /// Support answering: 1
    /// Do not support answering: 0
    public var supportAnswering: Bool
    /// Support mute: 1
    /// Do not support mute: 0
    public var supportMute: Bool
    /// Support hang up: 1
    /// Do not support hang up: 0
    public var supportHangUp: Bool
    /// Contact name (maximum 63 bytes)
    public var contact: String
    /// Phone number (maximum 31 bytes)
    public var phoneNumber: String
    /// Message content (maximum 249 bytes)
    public var dataText: String

    enum CodingKeys: String, CodingKey {
        case verison
        case evtType = "evt_type"
        case msgID = "msg_id"
        case supportAnswering = "support_answering"
        case supportMute = "support_mute"
        case supportHangUp = "support_hang_up"
        case contact
        case phoneNumber = "phone_number"
        case dataText = "data_text"
    }

    public init(evtType: Int, msgID: Int, supportAnswering: Bool, supportMute: Bool, supportHangUp: Bool, contact: String, phoneNumber: String, dataText: String) {
        self.verison = 0
        self.evtType = evtType
        self.msgID = msgID
        self.supportAnswering = supportAnswering
        self.supportMute = supportMute
        self.supportHangUp = supportHangUp
        self.contact = contact
        self.phoneNumber = phoneNumber
        self.dataText = dataText
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}
